common/djangoapps/track/middleware.py

```diff
@@ -204,3 +205,51 @@ def process_response(self, _request, response):
             pass
 
         return response
+
+
+    def truncate_openended_responses(self, event):
+        """
+        Truncates the tracking log entry for problems for which the
+        student response can be long winded.
+
+        event: dict, Constructed from POST and GET request params
+        """
+        event_json = json.loads(event)
+        event_post_keys = event_json['POST'].keys()
+        for key in event_post_keys:
+
+            # problem type with 'student_answer'
+            if key == 'student_answer' and isinstance(event_json['POST'].get('student_answer'), list):
+                truncated_answers = []
+                student_answer = event_json['POST'].get('student_answer')
+
+                # Loop round all the answers in the array and truncate each one, if necessary.
+                for answer in student_answer:
+                    truncated_answer = answer[:int(TRUNCATION_LENGTH/len(student_answer))]
```

Comment:
This is a bit dense. Can we split it out for readability?

```python
chunk_length = TRUNCATION_LENGTH / len(student_answer)
truncated_answer = answer[:chunk_length]
```

The `int` cast was unnecessary here, as we're declaring `TRUNCATION_LENGTH` to always be an `int` and `len` will also always return an `int`.